src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/FunctionResultContent.cs

```diff
@@ -20,10 +20,31 @@ public sealed class FunctionResultContent : AIContent
     /// </summary>
     /// <param name="callId">The function call ID for which this is the result.</param>
     /// <param name="name">The function name that produced the result.</param>
-    /// <param name="result">The function call result.</param>
-    /// <param name="exception">Any exception that occurred when invoking the function.</param>
+    /// <param name="result">
+    /// This may be <see langword="null"/> if the function returned <see langword="null"/>, if the function was void-returning
+    /// and thus had no result, or if the function call failed. Typically, however, in order to provide meaningfully representative
+    /// information to an AI service, a human-readable representation of those conditions should be supplied.
+    /// </param>
     [JsonConstructor]
-    public FunctionResultContent(string callId, string name, object? result = null, Exception? exception = null)
+    public FunctionResultContent(string callId, string name, object? result)
+    {
+        CallId = Throw.IfNull(callId);
+        Name = Throw.IfNull(name);
+        Result = result;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FunctionResultContent"/> class.
+    /// </summary>
+    /// <param name="callId">The function call ID for which this is the result.</param>
+    /// <param name="name">The function name that produced the result.</param>
+    /// <param name="result">
+    /// This may be <see langword="null"/> if the function returned <see langword="null"/>, if the function was void-returning
+    /// and thus had no result, or if the function call failed. Typically, however, in order to provide meaningfully representative
+    /// information to an AI service, a human-readable representation of those conditions should be supplied.
+    /// </param>
+    /// <param name="exception">Any exception that occurred when invoking the function.</param>
+    public FunctionResultContent(string callId, string name, object? result, Exception? exception)
```

Comment:
Why do we need to expose a ctor overload accepting Exception given that the property is settable?

Reply:
Trying to avoid breaking body changes right now if they're not necessary.

Reply:
Sure, but if we can't make them now it will be even more difficult to make them later.

Reply:
I'd rather take any such changes in batch when we have a meaningful set, so a more thorough review based on received feedback, etc. Force people to rev fewer times.

Reply:
Will we remember/have enough context to remove this constructor when such a review is going to be made? Unless we release new bits to customers on a very frequent basis I don't see how postponing such a decision would impact churn substantially.

Reply:
Each one of these is a binary breaking change. If every single release makes such a change but is binary breaking, it effectively means that the ecosystem gets reset with every single release.

Reply:
I understand, but at the same time this seems fair game or even expected when evolving previews.

Reply:
Removing this constructor is not a meaningful fix. It doesn't improve what someone is able to do. All it does is force someone to use a different syntax to construct it. Why would we want to reset the ecosystem for such a change? When there are other binary breaking changes we decide to make in preview, then we can make this at the same time, but I don't see the value in just making this one. And shipping will be frequent enough that it very well could be the only one.